src/components/TextLink.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {ForwardedRef, forwardRef, KeyboardEventHandler, MouseEventHandler, ReactElement} from 'react';
+import {GestureResponderEvent, Text as RNText, TextStyle} from 'react-native';
+import styles from '@styles/styles';
+import * as Link from '@userActions/Link';
+import CONST from '@src/CONST';
+import Text from './Text';
+
+type TextLinkProps = {
+    /** Link to open in new tab */
+    href: string;
+
+    /** Text content child */
+    children: ReactElement;
+
+    /** Additional style props */
+    style?: TextStyle;
+
+    /** Overwrites the default link behavior with a custom callback */
+    onPress?: () => void;
+
+    /** Callback that is called when mousedown is triggered */
+    onMouseDown?: MouseEventHandler;
+};
+
+function TextLink({href, children, style, onPress, onMouseDown = (event) => event.preventDefault(), ...props}: TextLinkProps, ref: ForwardedRef<RNText>) {
+    const openLink = () => {
+        if (onPress) {
+            onPress();
+            return;
+        }
+
+        Link.openExternalLink(href);
+    };
+
+    const openLinkOnTap = (event: GestureResponderEvent) => {
+        event.preventDefault();
+
+        openLink();
+    };
+
+    const openLinkOnEnterKey: KeyboardEventHandler = (event) => {
+        if (event.key !== 'Enter') {
+            return;
+        }
+        event.preventDefault();
+
+        openLink();
+    };
+
+    return (
+        <Text
+            style={[styles.link, style]}
+            role={CONST.ACCESSIBILITY_ROLE.LINK}
+            href={href}
+            onPress={openLinkOnTap}
+            onKeyDown={openLinkOnEnterKey}
+            onMouseDown={onMouseDown}
+            ref={ref}
+            suppressHighlighting
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
```

Comment:
`props.style` will overwrite former `style={[styles.link, style]}`

Reply:
That's how it used to be 🤷‍♂️ seems reasonable for some use cases. Do you want me to move `style={[styles.link, style]}` below `{...props}` @situchan ?

Reply:
Maybe you missed important line in current code:
https://github.com/Expensify/App/blob/b44f49efd4234e8ddca04525875b837c3994fbde/src/components/TextLink.js#L41

Reply:
Oh shoot, good call. Fixing